enhancements/installer/aws-custom-dns.md

```diff
@@ -0,0 +1,475 @@
+     installer adds IP addresses configured within the NLB or the DNS LB Name
+     configured in the ALB.]
+}
+```
+
+#### AWS Platform Struct within the Install Config [https://github.com/openshift/installer/blob/master/pkg/types/aws/platform.go]
+
+```go
+
+type CustomDNS struct {
+    // externallyManagedDNS is a string that needs to be explicitly set to
+    // "Enabled" when customer does not want the Installer to configure
+    // Route53 for them. The customer is responsible for responsible for
+    // configuring their own DNS solution based on LB configuration geberated
+    // by the Installer.
+    // Default value for this field is "Disabled". When "Disabled" the other
+    // fields in this struct will not be validated or used. 
+   
+    ExternallyManagedDNS string `json:"externallyManagedDNS"`
+
+    [TODO: Conclusion of the discussion in Design Details will determine if
+     admin needs to provide IP addresses.]
+}
+
+type Platform struct {
+// <snip>
+
+// All configuration required for enabling the optional customDNS
+CustomDNS CustomDNS `json:"customDNS,omitempty"`
+}
+```
+
+### Implementation Details/Notes/Constraints [optional]
+
+
+### Risks and Mitigations
+
+
+### Drawbacks
+
+## Design Details
+
+### Open Questions [optional]
+
+One of the decisions that needs to be made early in the design process is to
+determine the best way to order OpenShift's configuration of the Load Balancer
+and the customer's configuration of their custom DNS solution.
+
+Today, the Installer configures the API's LB first and the generated DNS name
+is used to configure DNS CNAME records. The LB DNS name cannot be predicted in
+advance because it is generated using an unique UUID while configuring LB
+entries for API, API-Int and Ingress services.
+
+Here are a few options being considered as potential solutions:
+
+#### Option 1: Custom DNS configuration before creating the cluster
+
+In this Option, the admin manually configures the Custom DNS solution before
+cluster creation is started. The Custom DNS is configured with an A record
+that points to static IP addresses within the public subnets of the machine
+network.
+
+When users provide their own subnets during install, they are responsible
+for creating the DNS records for the API servers using the IP addresses for
+NICs on the pre-created public subnets. Similarly, the IP addresses for NICs
+on the private subnets can be provided to the MCO via the Infrastruture CR.
+MCO uses these IP addresses to configure DNS for the Internal API Service.
+
+Is there a scenario when the public and private subnets are not created in
+advance? In that case, should the user be providing the API and API-Int IP
+addresses via install-config?
+
+This [RFE](https://access.redhat.com/solutions/4846611) talks about only using
+CNAME records for DNS configuration. How should this option be modified to
+handle this RFE?
+
+#### Option 2: Custom DNS configuration after creating the cluster
```

Comment:
The Installer host needs to be able to connect to the API. It is unclear how that would happen in this option.

Reply:
Yes, the initial understanding was that only the API-Int Server needs to be accessible during cluster installation. Since then, I have learnt that api, api-int and *.apps should all be resolvable during cluster install. Design has been updated to handle that.